src/types.rs

```diff
@@ -1762,3 +1762,582 @@ impl std::fmt::Display for BasinName {
+        api::Operation::try_from(value)
+            .map_err(|_| "invalid operation".into())
+            .map(Into::into)
+    }
+}
+
+impl From<Operation> for api::Operation {
+    fn from(value: Operation) -> Self {
+        match value {
+            Operation::Unspecified => Self::Unspecified,
+            Operation::ListBasins => Self::ListBasins,
+            Operation::CreateBasin => Self::CreateBasin,
+            Operation::DeleteBasin => Self::DeleteBasin,
+            Operation::ReconfigureBasin => Self::ReconfigureBasin,
+            Operation::GetBasinConfig => Self::GetBasinConfig,
+            Operation::IssueAccessToken => Self::IssueAccessToken,
+            Operation::RevokeAccessToken => Self::RevokeAccessToken,
+            Operation::ListAccessTokens => Self::ListAccessTokens,
+            Operation::ListStreams => Self::ListStreams,
+            Operation::CreateStream => Self::CreateStream,
+            Operation::DeleteStream => Self::DeleteStream,
+            Operation::GetStreamConfig => Self::GetStreamConfig,
+            Operation::ReconfigureStream => Self::ReconfigureStream,
+            Operation::CheckTail => Self::CheckTail,
+            Operation::Append => Self::Append,
+            Operation::Read => Self::Read,
+            Operation::Trim => Self::Trim,
+            Operation::Fence => Self::Fence,
+        }
+    }
+}
+
+impl From<api::Operation> for Operation {
+    fn from(value: api::Operation) -> Self {
+        match value {
+            api::Operation::Unspecified => Self::Unspecified,
+            api::Operation::ListBasins => Self::ListBasins,
+            api::Operation::CreateBasin => Self::CreateBasin,
+            api::Operation::DeleteBasin => Self::DeleteBasin,
+            api::Operation::ReconfigureBasin => Self::ReconfigureBasin,
+            api::Operation::GetBasinConfig => Self::GetBasinConfig,
+            api::Operation::IssueAccessToken => Self::IssueAccessToken,
+            api::Operation::RevokeAccessToken => Self::RevokeAccessToken,
+            api::Operation::ListAccessTokens => Self::ListAccessTokens,
+            api::Operation::ListStreams => Self::ListStreams,
+            api::Operation::CreateStream => Self::CreateStream,
+            api::Operation::DeleteStream => Self::DeleteStream,
+            api::Operation::GetStreamConfig => Self::GetStreamConfig,
+            api::Operation::ReconfigureStream => Self::ReconfigureStream,
+            api::Operation::CheckTail => Self::CheckTail,
+            api::Operation::Append => Self::Append,
+            api::Operation::Read => Self::Read,
+            api::Operation::Trim => Self::Trim,
+            api::Operation::Fence => Self::Fence,
+        }
+    }
+}
+
+#[sync_docs]
+#[derive(Debug, Clone, Default)]
+pub struct AccessTokenScope {
+    pub basins: Option<ResourceSet>,
+    pub streams: Option<ResourceSet>,
+    pub tokens: Option<ResourceSet>,
+    pub op_groups: Option<PermittedOperationGroups>,
```

Comment:
Are these fields really optional? Doesn't seem so looking at the proto definitions. Let's have apt `Default` impls instead of using optionals then?